engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/task/UserTaskProcessor.java

```diff
@@ -0,0 +1,135 @@
+    } else {
+      throw new BpmnProcessingException(
+          context, "Expected to process user task, but could not determine processing behavior");
+    }
+  }
+
+  private static final class ZeebeUserTaskBehavior implements UserTaskBehavior {
+
+    private final BpmnIncidentBehavior incidentBehavior;
+    private final BpmnStateTransitionBehavior stateTransitionBehavior;
+    private final BpmnVariableMappingBehavior variableMappingBehavior;
+    private final BpmnEventSubscriptionBehavior eventSubscriptionBehavior;
+    private final BpmnStateBehavior stateBehavior;
+    private final ExpressionProcessor expressionProcessor;
+
+    private final EventTriggerBehavior eventTriggerBehavior;
+
+    public ZeebeUserTaskBehavior(
+        final BpmnBehaviors bpmnBehaviors,
+        final BpmnStateTransitionBehavior stateTransitionBehavior) {
+      eventSubscriptionBehavior = bpmnBehaviors.eventSubscriptionBehavior();
+      incidentBehavior = bpmnBehaviors.incidentBehavior();
+      this.stateTransitionBehavior = stateTransitionBehavior;
+      variableMappingBehavior = bpmnBehaviors.variableMappingBehavior();
+      stateBehavior = bpmnBehaviors.stateBehavior();
+      expressionProcessor = bpmnBehaviors.expressionBehavior();
+      eventTriggerBehavior = bpmnBehaviors.eventTriggerBehavior();
+    }
+
+    @Override
+    public void onActivate(final ExecutableUserTask element, final BpmnElementContext context) {}
+
+    @Override
+    public void onComplete(final ExecutableUserTask element, final BpmnElementContext context) {}
+
+    @Override
+    public void onTerminate(final ExecutableUserTask element, final BpmnElementContext context) {}
+  }
+
+  private static final class JobWorkerTaskBehavior implements UserTaskBehavior {
+
+    private final JobWorkerTaskProcessor delegate;
+
+    public JobWorkerTaskBehavior(
+        final BpmnBehaviors bpmnBehaviors,
+        final BpmnStateTransitionBehavior stateTransitionBehavior) {
+      delegate = new JobWorkerTaskProcessor(bpmnBehaviors, stateTransitionBehavior);
+    }
+
+    @Override
+    public void onActivate(final ExecutableUserTask element, final BpmnElementContext activating) {
+      delegate.onActivate(element, activating);
+    }
+
+    @Override
+    public void onComplete(final ExecutableUserTask element, final BpmnElementContext completing) {
+      delegate.onComplete(element, completing);
+    }
+
+    @Override
+    public void onTerminate(
+        final ExecutableUserTask element, final BpmnElementContext terminating) {
+      delegate.onTerminate(element, terminating);
+    }
+  }
+
+  /** Extract different behaviors depending on the type of task. */
+  private interface UserTaskBehavior {
+    void onActivate(ExecutableUserTask element, BpmnElementContext activating);
+
+    void onComplete(ExecutableUserTask element, BpmnElementContext completing);
+
+    void onTerminate(ExecutableUserTask element, BpmnElementContext terminating);
+  }
```

Comment:
:wrench: I think it would be nice if we could reduce the duplication of this code. It repeats (almost exactly the same) in the `BusinessRuleTaskProcessor` and `ScriptTaskProcessor`. The only difference is the input argument for the `onActivate/onComplete/onTerminate` methods.

We can extract this code in a separate file, and we can use generics to specifiy the `ExecutableUserTask`, `ExecutableScriptTask`, and `ExecutableBusinessRuleTask`.

Maybe we can call it `TaskBehavior<T extends ExecutableJobWorkerTask>` where the `T` would be set to `ExecutableUserTask`, `ExecutableScriptTask`, or `ExecutableBusinessRuleTask` depending on the implementation.

The `JobWorkerTaskBehavior` class can also be extracted, and I guess implementing `TaskBehavior<ExecutableJobWorkerTask>` would be enough.

What do you think @tmetzke?

Reply:
I wasn't all that fond of duplicating this as well but I didn't have any nice idea on the spot to solve this elegantly. I will give this a try, sounds nice 👍 If I get stuck or it doesn't work as expected, we can reiterate together 🙂

Reply:
I took a slightly different approach to simplify this a bit more. Let me know what you think.